packages/kbn-text-based-editor/src/history_local_storage.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import moment from 'moment';
+import 'moment-timezone';
+const QUERY_HISTORY_ITEM_KEY = 'QUERY_HISTORY_ITEM_KEY';
+const dateFormat = 'MMM. D, YY HH:mm:ss.SSS';
+
+/**
+ * We show maximum 20 ES|QL queries in the Query history component
+ */
+
+export interface QueryHistoryItem {
+  status?: 'success' | 'error' | 'warning';
+  queryString: string;
+  startDateMilliseconds?: number;
+  timeRan?: string;
+  timeZone?: string;
+  duration?: string;
+  queryRunning?: boolean;
+}
+
+const MAX_QUERIES_NUMBER = 20;
+
+const getKey = (queryString: string) => {
+  return queryString.replaceAll('\n', '').trim();
```

Comment:
Do we want to make this more robust? Right now, internal whitespace changes lead to a new entry (except adding pure newlines).

https://github.com/elastic/kibana/assets/315764/d5c06d3b-a667-4d71-9c03-8770f6003170

This behavior also shows up for our "Add line breaks on pipes" quick action because it removes the trailing spaces of each line:

https://github.com/elastic/kibana/assets/315764/64ffaba3-f7a6-45c4-ad4c-ef3b70648903